src/components/DownloadImageDialog/version.ts

```diff
@@ -0,0 +1,113 @@
+import uniq from 'lodash/uniq';
+import partition from 'lodash/partition';
+import { Dictionary, OsVersion } from './models';
+
+export type VersionSelectionOptions = {
+	title: string;
+	value: string;
+	isRecommended?: boolean;
+	osType: string;
+	line?: string;
+	knownIssueList: string | null;
+} & (
+	| {
+			hasPrebuiltVariants: false;
+			rawVersion: string;
+	  }
+	| {
+			hasPrebuiltVariants: true;
+			rawVersions: {
+				dev?: string;
+				prod?: string;
+			};
+	  }
+);
+
+export const transformVersions = (versions: OsVersion[]) => {
+	// Get a single object per stripped version, with both variants of it included (if they exist). It expects a sorted `
+	const optsByVersion: Dictionary<VersionSelectionOptions> = {};
+	versions.forEach((version) => {
+		const existingSelectionOpt = optsByVersion[version.strippedVersion];
+		// We always want to use the 'prod' variant's formatted version as it can contain additional information (such as recommended label).
+		const title =
+			(version.variant === 'dev' ? existingSelectionOpt?.title : null) ??
+			version.strippedVersion;
+
+		optsByVersion[version.strippedVersion] = {
+			title,
+			value: version.strippedVersion,
+			osType: version.osType,
+			line: version.line,
+			isRecommended: version.isRecommended,
+			knownIssueList: version.known_issue_list,
+			// Unified releases in the model have variant === ''
+			// but we also test for nullish for backgwards compatibility w/ the typings.
+			...(!version.variant
+				? {
+						hasPrebuiltVariants: false,
+						rawVersion: version.raw_version,
+				  }
+				: {
+						hasPrebuiltVariants: true,
+						rawVersions: {
+							...(existingSelectionOpt != null &&
+								'rawVersions' in existingSelectionOpt &&
+								existingSelectionOpt.rawVersions),
+							[version.variant]: version.raw_version,
+						},
+				  }),
+		};
+	});
+
+	return Object.values(optsByVersion);
+};
+
+const LEGACY_OS_VERSION_MAJOR = 1;
+
+// This returns the 3 most preferred versions for an os type. For multi-line os types, that would be the latest of each line, otherwise it is the latest 3 versions.
+export const getPreferredVersionOpts = (
+	versionOpts: VersionSelectionOptions[],
+) => {
+	const [supportedVersions, legacyVersions] = partition(versionOpts, (v) => {
+		// TODO: check if worth installing semver on ui-shared-components;
```

Comment:
Curious, why might it not be worth it?

Reply:
not sure it is valuable to install an entire library to just remove a single line of code. I would first check if we can also use it to improve other pieces.

Reply:
I was thinking it might count as a shared component since we use it in the UI, but I see now we don't use it in hub so it doesn't count as shared yet